pkgs/games/vassal/default.nix

```diff
@@ -1,19 +1,21 @@
-{ stdenv, fetchurl, jre, makeWrapper }:
+{ stdenv, fetchzip, jre, makeWrapper }:
 
 stdenv.mkDerivation rec {
-  name = "VASSAL-3.3.2";
+  pname = "VASSAL";
+  version = "3.4.12";
+  name = "${pname}-${version}";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/vassalengine/${name}-linux.tar.bz2";
-    sha256 = "1abhlkl27gyfa1lghvv76xa6ks5hiwv2s9wb9ddadm0m07f87n1w";
+  src = fetchzip {
+    url = "https://github.com/vassalengine/vassal/releases/download/${version}/${pname}-linux.tar.bz2";
```

Comment:
This does not appear to the the right link:

```
builder for '/nix/store/5mw82j19kbzrz7lvifgf81i032lqvn5y-source.drv' failed with exit code 1; last 7 log lines:

  trying https://github.com/vassalengine/vassal/releases/download/3.4.12/VASSAL-linux.tar.bz2
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
    0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
  curl: (22) The requested URL returned error: 404 Not Found
  error: cannot download source from any mirror
cannot build derivation '/nix/store/a0m5yywi3mi6p7rwj2xkf4aa53wxcdzf-VASSAL-3.4.12.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bj7q2mxbr078s2s4sp946fh8dp89h9xs-env.drv': 1 dependencies couldn't be built
[0 built (1 failed), 119 copied (1041.6 MiB), 484.3 MiB DL]
error: build of '/nix/store/bj7q2mxbr078s2s4sp946fh8dp89h9xs-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/107897
1 package failed to build:
vassal
```

If I check the releases under GitHub then I see a different link:
https://github.com/vassalengine/vassal/archive/3.4.12.tar.gz

Reply:
I didn't see that it's using the prebuilt linux, so the above comment is not relevant.

However the `${pname}-linux.tar.bz2` should be `${name}-linux.tar.bz2`

Also the whole thing doesn't builds for me.

Reply:
Uh, sorry. This was because I hade the source already in the store and the sha didn't change when I rewrote the URL to use variables.

I have deleted the source and the package from the store and rebuilt, now seems to work.